reVX/utilities/forecasts.py

```diff
@@ -0,0 +1,386 @@
+import os
+import shutil
+from warnings import warn
+
+from rex import Resource
+
+logger = logging.getLogger(__name__)
+
+
+class Forecasts:
+    """
+    Forecast Utility
+    """
+    def __init__(self, fcst_h5, fcst_dset, actuals_h5=None, actuals_dset=None):
+        """
+        Parameters
+        ----------
+        fcst_h5 : str
+            Path to forecast .h5 file
+        fcst_dset : str
+            Dataset to correct
+        actuals_h5 : str, optional
+            Path to forecast to .h5 file, by default None
+        actuals_dset : str, optional
+            Actuals dataset, by default None
+        """
+        self._fcst_h5 = fcst_h5
+        if actuals_h5 is None:
+            actuals_h5 = fcst_h5
+
+        self._actuals_h5 = actuals_h5
+
+        self._fcst_dset = fcst_dset
+
+        if actuals_dset is None:
+            actuals_dset = fcst_dset
+
+        self._actuals_dset = actuals_dset
+
+        self._a_slice = self._preflight_check()
+
+    @property
+    def fcst_h5(self):
+        """
+        Path to forecast .h5 file
+
+        Returns
+        -------
+        str
+        """
+        return self._fcst_h5
+
+    @property
+    def actuals_h5(self):
+        """
+        Path to actuals .h5 file
+
+        Returns
+        -------
+        str
+        """
+        return self._actuals_h5
+
+    @property
+    def fcst_dset(self):
+        """
+        Forecast dset to correct
+
+        Returns
+        -------
+        str
+        """
+        return self._fcst_dset
+
+    @property
+    def actuals_dset(self):
+        """
+        Actuals dset to use to correct forecasts
+
+        Returns
+        -------
+        str
+        """
+        return self._actuals_dset
+
+    @staticmethod
+    def bias_correct_fcst(actuals, fcsts):
+        """
+        Bias correct forecast data
+
+        Parameters
+        ----------
+        actuals : ndarray
+            Timeseries actuals (time x sites)
+        fcsts : ndarray
+            Timeseries forecats (time x sites)
+
+
+        Returns
+        -------
+        fcsts : ndarray
+            Bias corrected forecasts
+        """
+        bc_factors = actuals.sum(axis=0) / fcsts.sum(axis=0)
+        fcsts = fcsts * bc_factors
+        actuals_max = actuals.max(axis=0)
+        mask = fcsts >= actuals_max
+        fcsts = np.where(mask, actuals_max, fcsts)
+
+        return fcsts
+
+    @classmethod
+    def blend_fcsts(cls, actuals, fcsts, fcst_perc):
+        """
+        Bias correct and blend forecasts with actuals
+
+        Parameters
+        ----------
+        actuals : ndarray
+            Timeseries actuals (time x sites)
+        fcsts : ndarray
+            Timeseries forecats (time x sites)
+        fcst_perc : float
+            Percentage of forecast to use for blending
+
+        Returns
+        -------
+        fcsts : ndarray
+            Bias corrected and blended forecasts
+        """
+        fcsts = cls.bias_correct_fcst(actuals, fcsts)
```

Comment:
I think you should include in both the method doc strings and the class doc string the exact steps taken to manipulate the forecast data. This will help with transparent documentation propagated through to the docs.

Reply:
Good call